newIDE/app/src/MainFrame/index.js

```diff
@@ -1163,16 +1166,19 @@ class MainFrame extends React.Component<Props, State> {
   };
 
   save = () => {
+    this.setState({
```

Comment:
I don't think you need this. Also, calling setState can be asynchronous, meaning that as you're reading `this.state.isSaveAs` just after, you can get the old value (true)!

Only use setState when it's something that must be shown on screen by React.

Reply:
Actually, see the following comments, you can remove the modifications made to this function I think.